analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -750,11 +750,17 @@ public void Invocation_InformationIsNothing_NoTrackedSymbol()
         }
 
         [DataTestMethod]
-        [DataRow("arg != null")]
-        [DataRow("arg is not null")]
-        [DataRow("arg is { }")]
-        public void Invocation_DebugAssert_LearnsNotNull_Simple(string expression) =>
-            DebugAssertValues(expression).Should().HaveCount(1).And.ContainSingle(x => x.HasConstraint(ObjectConstraint.NotNull));
+        [DataRow("arg != null", "object")]
+        [DataRow("arg != null", "int?")]
+        [DataRow("arg != null", "bool?")]
```

Comment:
Would be good to have such tests as well to see if all constraints are applied and propagated correctly:
```cs
        [DataRow("arg", "bool?")] //NotNull, True
        [DataRow("arg is false", "bool?")] //NotNull, False
        [DataRow("arg is not true", "bool?")] // No constraint
```

Reply:
I've added `GetValueOrDefault`. The `??` splits in two states, without saying anything useful - it would add too much scaffolding with nothing to really assert.